src/index.js

```diff
@@ -13,22 +14,34 @@ const defaultSettings = {
     },
 };
 
-function getEnvironment() {
+function getEnvIndex(args) {
     const envFlags = ['-e', '--env'];
-    const args = process.argv;
-    const index = args.findIndex(arg => envFlags.includes(arg)) + 1;
+    return args.findIndex(arg => envFlags.includes(arg)) + 1;
+};
+
+function getEnvironment(args) {
+    const index = getEnvIndex(args);
 
-    return (index > 0 && index < args.length) ? process.argv[index] : 'default';
-}
+    return (index > 0 && index < args.length) ? args[index] : 'default';
+};
 
 module.exports = class Deviance {
-    constructor(settings) {
+    constructor(settings = {}, args = process.argv) {
```

Comment:
I'm not adverse to this, but I do wonder if creating a param here invites/suggests it should be overwritten. Could Jest not just set `process.argv.push('whatever');` before the tests that require it?

Reply:
I've updated to use ```process.argv.push('-e', 'value')``` when testing deviance but we still require the argument to test getEnvironments function.